RepoDb.Core/RepoDb/Cachers/DbFieldCache.cs

```diff
@@ -191,10 +215,31 @@ public static class DbFieldCache
             if (cache.TryGetValue(key, out var result) == false)
             {
                 // Get from DB
-                result = new DbFieldCollection(await connection
-                    .GetDbHelper()
-                    .GetFieldsAsync(connection, tableName, transaction, cancellationToken),
-                    connection.GetDbSetting());
+                var dbHelper = connection.GetDbHelper();
+                var semaphore = semaphores.GetOrAdd(dbHelper, _ => new SemaphoreSlim(1, 1));
+                Debug.WriteLine($"Getting fields async for '{tableName}' from {connection.GetType()}");
+                await semaphore.WaitAsync();
```

Comment:
Probably we can lock only for PostgreSQL as SQL Server supports multiple active results

Reply:
SQL server only allows multiple active record sets when enabled in the connection string (via internal multiplexing); so it is not always enabled and is often disabled.